### PR TITLE
Remove Philox/Generator dependency in flash_sparse_api.cpp

### PR DESCRIPTION
**Pull request by janeyx99**
We want to migrate sglang/sgl-kernel to ABI stable (https://github.com/sgl-project/sglang/issues/26598) and sgl-kernel only pulls in the flash_sparse_api.cpp (and some .cu files) from this fork.

PyTorch doesn't have stable ABI equivalents for Philox/at::Generator version 2.11 (desired target runtime version). These APIs are thankfully only used for dropout, which is commented out dead code in the file as inference doesn't use dropout.

This change does 2 things:
1. It deletes the commented out dead code completely, as well as now-dead references to Generator and Philox APIs in flash_sparse_api.cpp. We then recursively delete the now unreachable dropout code in the sparse launch template, which includes eliminating return_softmax (only enabled if dropout).
2. This one I have a question to the maintainers: for stable ABI migration, we would need to detangle flash_sparse_api.cpp from pulling in the unstable APIs (at::Generator or the PhiloxState from ATen/cuda/CUDAGeneratorImpl.h). Currently, those are transitively included from flash_sparse.h pulling in flash.h. **In order to minimize deviance from the original code, I've opted for swapping out the state with an opaque buffer.** Another option is creating an entirely different kernel arg struct for the sparse APIs, but that would induce more code changes across the kernel code, but could be a clearer alternative. Some important context is that flash_sparse_api.cpp is the only file fetched into sgl-kernel relating to FA2: https://github.com/sgl-project/sglang/blob/main/sgl-kernel/CMakeLists.txt#L313-L317 so I am suspecting very few people use this fork's flash_api.cpp anyway.

Test plan:
Locally using scaffolding in https://github.com/sgl-project/sglang/pull/26827 to pull in, build and test this change. 
- [x] built
- [x] `pytest --noconftest tests/test_flash_attention.py` all pass

### CODE REVIEW

File: csrc/flash_attn/src/flash.h [hunk 1]
```diff
@@ -118,8 +117,12 @@ struct Flash_fwd_params : public Qkv_params {
...
     float softcap;
 
-    // Random state.
-    at::PhiloxCudaState philox_args;
+    // Opaque storage for at::PhiloxCudaState. Decouples this header (and the
+    // sparse inference path that transitively includes it) from
+    // <ATen/cuda/CUDAGeneratorImpl.h>. Training code in flash_api.cpp writes
+    // it via placement-new; kernels read it via reinterpret_cast. Size and
+    // alignment are checked with a static_assert in flash_api.cpp.
+    alignas(8) unsigned char philox_args[32];
```
Comment (janeyx99) [inline csrc/flash_attn/src/flash.h:125]: To explain why this buffer, the PhiloxCudaState struct is 24 bytes, which we round up here for some cushion:

```
struct PhiloxCudaState {
    union Payload { uint64_t val; int64_t* ptr; };  // 8 bytes
    Payload seed_;                                   // 8
    Payload offset_;                                 // 8
    uint32_t offset_intragraph_ = 0;                 // 4
    bool captured_ = false;                          // 1 (+3 tail padding to keep struct alignment 8)
  };
```

File: csrc/flash_attn/flash_api.cpp [hunk 2]
```diff
@@ -476,6 +483,8 @@ mha_fwd_sparse(at::Tensor &q,         // batch_size x seqlen_q x num_heads x hea
     TORCH_CHECK(head_size_og <= 256, "FlashAttention forward only supports head dimension at most 256");
     TORCH_CHECK(num_heads % num_heads_k == 0, "Number of heads in key/value must divide number of heads in query");
 
+    TORCH_CHECK(p_dropout == 0.0f, "Sparse attention does not support dropout for inference");
```
Comment (janeyx99) [inline csrc/flash_attn/flash_api.cpp:486]: i also added new checks
